worlds/mm2/client.py

```diff
@@ -0,0 +1,562 @@
+                                   f"Please use that version to connect instead.\n"
+                                   f"Patch version: ({older_version})\n"
+                                   f"Client version: ({'.'.join([str(i) for i in MM2World.world_version])})")
+                if "pool" in ctx.command_processor.commands:
+                    ctx.command_processor.commands.pop("pool")
+                if "request" in ctx.command_processor.commands:
+                    ctx.command_processor.commands.pop("request")
+                if "autoheal" in ctx.command_processor.commands:
+                    ctx.command_processor.commands.pop("autoheal")
+                return False
+        except UnicodeDecodeError:
+            return False
+        except RequestFailedError:
+            return False  # Should verify on the next pass
+
+        ctx.game = self.game
+        self.rom = game_name
+        ctx.items_handling = 0b111
+        ctx.want_slot_data = False
+        deathlink = (await read(ctx.bizhawk_ctx, [(0x3FFC5, 1, "PRG ROM")]))[0][0]
+        if deathlink & 0x01:
+            self.death_link = True
+        if deathlink & 0x02:
+            self.energy_link = True
+
+        if self.energy_link:
+            if "pool" not in ctx.command_processor.commands:
+                ctx.command_processor.commands["pool"] = cmd_pool
+            if "request" not in ctx.command_processor.commands:
+                ctx.command_processor.commands["request"] = cmd_request
+            if "autoheal" not in ctx.command_processor.commands:
+                ctx.command_processor.commands["autoheal"] = cmd_autoheal
+
+        return True
+
+    async def set_auth(self, ctx: "BizHawkClientContext") -> None:
+        if self.rom:
+            ctx.auth = b64encode(self.rom).decode()
+
+    def on_package(self, ctx: "BizHawkClientContext", cmd: str, args: Dict[str, Any]) -> None:
+        if cmd == "Bounced":
+            if "tags" in args:
+                assert ctx.slot is not None
+                if "DeathLink" in args["tags"] and args["data"]["source"] != ctx.slot_info[ctx.slot].name:
+                    self.on_deathlink(ctx)
+        elif cmd == "Retrieved":
+            if f"MM2_LAST_WILY_{ctx.team}_{ctx.slot}" in args["keys"]:
+                self.last_wily = args["keys"][f"MM2_LAST_WILY_{ctx.team}_{ctx.slot}"]
+        elif cmd == "Connected":
+            if self.energy_link:
+                ctx.set_notify(f"EnergyLink{ctx.team}")
+                if ctx.ui:
+                    ctx.ui.enable_energy_link()
+
+    async def send_deathlink(self, ctx: "BizHawkClientContext") -> None:
+        self.sending_death_link = True
+        ctx.last_death_link = time.time()
+        await ctx.send_death("Mega Man was defeated.")
+
+    def on_deathlink(self, ctx: "BizHawkClientContext") -> None:
+        ctx.last_death_link = time.time()
+        self.pending_death_link = True
+
+    async def game_watcher(self, ctx: "BizHawkClientContext") -> None:
```

Comment:
This method is really big. I think that doing some sub-method would make the code easier to follow.